pkg/volume/csi/csi_plugin.go

```diff
@@ -123,9 +123,7 @@ func (h *RegistrationHandler) ValidatePlugin(pluginName string, endpoint string,
 		// The deprecated dir will only be allowed for a whitelisted set of old versions.
 		// CSI 1.x drivers should use the /var/lib/kubelet/plugins_registry
 		if !isDeprecatedSocketDirAllowed(versions) {
-			err := fmt.Errorf("socket for CSI driver %q versions %v was found in a deprecated dir. Drivers implementing CSI 1.x+ must use the new dir", pluginName, versions)
-			klog.Error(err)
-			return err
+			return fmt.Errorf("socket for CSI driver %q versions %v was found in a deprecated dir. Drivers implementing CSI 1.x+ must use the new dir", pluginName, versions)
```

Comment:
When should we use `fmt.Errorf` vs `errors.New`?

Reply:
I think it basiclly the same thing, when we need to format the output string,  fmt.Errorf is better in use, when it just raw string, erros.New is ok. fmt.Errof internally called errors.New after formatting.

Reply:
errors.New(log()) = fixed-csi-info-string + fmt.Errorf(). 
Shoud we use errors.New(log()) for uniform format?

Reply:
Yes I think that is a good idea

Reply:
Got it, thank you for your suggestion.